src/app/components/Input/InputBase.tsx

```diff
@@ -0,0 +1,65 @@
+import React, {
+  useState,
+  HTMLProps,
+  useCallback,
+  useEffect,
+  ChangeEvent,
+  ChangeEventHandler,
+} from 'react';
+import debounceCallback from 'lodash.debounce';
+
+export type InputBaseProps = Omit<HTMLProps<HTMLInputElement>, 'ref'> & {
+  debounce?: number;
+  dataActionId?: string;
+  onChangeText?: (value: string) => void;
+  /** @deprecated Use onChangeText if possible */
+  onChange?: ChangeEventHandler<HTMLInputElement>;
+};
+
+export const InputBase = React.forwardRef<HTMLInputElement, InputBaseProps>(
+  (
+    { value, debounce = 500, dataActionId, onChange, onChangeText, ...props },
+    ref,
+  ) => {
+    const [renderedValue, setRenderedValue] = useState<
+      string | string[] | number | undefined
+    >(value);
+
+    // inter wants to have debounceCallback as dependency, but it would create memory leaks
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const debouncedOnChangeHandler = useCallback(
```

Comment:
You can just use useMemo and exhaustive deps analysis should work fine.
```suggestion
    const debouncedOnChangeHandler = useMemo( () =>
```